controlplane/controllers/ck8scontrolplane_controller.go

```diff
@@ -738,10 +738,25 @@ func (r *CK8sControlPlaneReconciler) upgradeControlPlane(
 		return reconcile.Result{}, fmt.Errorf("failed to upgrade kubelet config map")
 	}
 	**/
+	logger := r.Log.WithValues("namespace", kcp.Namespace, "CK8sControlPlane", kcp.Name, "cluster", cluster.Name)
 
-	if controlPlane.Machines.Len() <= int(*kcp.Spec.Replicas) {
-		// scaleUp ensures that we don't continue scaling up while waiting for Machines to have NodeRefs
-		return r.scaleUpControlPlane(ctx, cluster, kcp, controlPlane)
+	if kcp.Spec.RolloutStrategy == nil {
+		logger.Info("RolloutStrategy is empty, unable to continue")
+		return ctrl.Result{}, nil
+	}
+
+	switch kcp.Spec.RolloutStrategy.Type {
```

Comment:
I am getting here a `"panicGoValue": "\"invalid memory address or nil pointer dereference\"",` -> which means I am not doing the correct defaulting. Is there something I am missing?

Reply:
This has been resolved by adding the default value via annotations.